src/aio.c

```diff
@@ -195,6 +196,21 @@ static void isaio_complete(void *arg) {
 
 }
 
+
+static void raio_invoke_cb(void* arg) {
+  nano_aio *raio = (nano_aio *) arg;
+  if (raio->cb == NULL || Rf_isNull(raio->cb)) return;
+  SEXP func = (SEXP)raio->cb;
+  SEXP callExpr, result;
+  if (!Rf_isNull(func)) {
+    PROTECT(callExpr = Rf_lcons(func, R_NilValue)); // Prepare call
+    PROTECT(result = Rf_eval(callExpr, R_GlobalEnv)); // Execute call
+
+    UNPROTECT(2);
+    R_ReleaseObject(func);
```

Comment:
Not super confident in this part, co-wrote with ChatGPT. Maybe should be written not to `PROTECT` the result since we don't use it?

Reply:
Time to start using ChatGPT! You're right we don't need to protect the result, or we could even cast the `Rf_eval` to `void`. More importantly, we shouldn't be adding/removing potentially large numbers of objects to/from the R precious list - but that's a detail that can be solved.

Reply:
Actually I don't think we need to protect the callback functions in the Precious List, which simplifies things greatly. I've continued your work in the 'callbacks' branch as I don't seem to have permissions to modify this PR. I'm still thinking through what an optimal API would be.